### PR TITLE
Student merging backend

### PR DESCRIPTION
**Pull request by antonlaukemper**
I changed something in the beforeEach of the surveyController specs which led to an interesting error that needs to be investigated:
grafik

(BTW thank you @rauwuckl  for putting in those logs or this would have failed silently!!)
I am assuming this is a minor error somewhere in the tests, but it should still not be ignored

### CODE REVIEW

File: Backend/scala-backend/app/controllers/SurveyController.scala [hunk 1]
```diff
@@ -146,18 +172,119 @@ class SurveyController @Inject() (
...
+      )
+
+    val creationSuccess: Future[Boolean] = Future
+      .sequence(creationSuccessList)
+      .map((arr: Seq[Boolean]) => {
+        arr.forall(_ == true)
+      })
+
+    // 2. rename
+    creationSuccess.flatMap(cSuccess => {
```
Comment (rauwuckl) [inline Backend/scala-backend/app/controllers/SurveyController.scala:228]: flatten out
on the LHS of 228: val renameSuccess: Future[Boolean] = creationSucess.flatMap(...
...
return on line 238
})

File: Backend/scala-backend/app/models/StudentModel.scala [hunk 2]
```diff
@@ -69,8 +70,45 @@ class StudentModel(db: Database)(implicit ec: ExecutionContext) {
...
+          entry.encryptedname,
+          entry.selfreported,
+          entry.groupbelonging
+        )))
+  }
+
   def updateGroupBelonging(studentId: Int, group: Int): Future[Int] = {
     val updateOp: Future[Int] = db.run(Student.filter(_.id === studentId).map(row => (row.groupbelonging)).update((Some(group))))
+    
+    updateOp.map(x => println(x))
```
Comment (rauwuckl) [inline Backend/scala-backend/app/models/StudentModel.scala:96]: print

**Review by rauwuckl — CHANGES_REQUESTED** [review]
Amazing tests. Love it so much <3

**Comment by antonlaukemper** [comment]
Ein dutzend neuer commits :D

**Review by rauwuckl — APPROVED** [review]
LGTM

**Pull request merged**